### PR TITLE
Paginated Merchant Transactions

### PR DESCRIPTION
**Pull request by husobee**
fixes: #265 - updates to allow for pagination and implementation of merchant transactions

### CODE REVIEW

File: payment/datastore.go [hunk 1]
```diff
@@ -143,6 +146,64 @@ func (pg *Postgres) GetOrder(orderID uuid.UUID) (*Order, error) {
...
+	count = len(v) + start
+	if orderBy != "" {
+		getStatement += orderBy
+		params = append(params, v...)
+	}
+
+	offset := pagenation.Page * pagenation.Items
+	if offset > 0 {
+		getStatement += fmt.Sprintf(" OFFSET $%d", count)
+		count++
```
Comment (corymcdonald) [inline payment/datastore.go]: What's the incrementation for?

Comment (husobee) [reply payment/datastore.go]: We have no idea how the caller may wish to order by, so we need to dynamically structure the sql query to accommodate.  The count is the parameter count within the query `$1 $2` etc.

File: payment/datastore.go [hunk 2]
```diff
@@ -143,6 +146,75 @@ func (pg *Postgres) GetOrder(orderID uuid.UUID) (*Order, error) {
...
+		`
+
+	// $ numbered params for query
+	params := []interface{}{
+		merchantID,
+	}
+
+	orderBy := pagination.GetOrderBy()
+	if orderBy != "" {
+		getStatement += fmt.Sprintf(" ORDER BY %s", orderBy)
```
Comment (corymcdonald) [inline payment/datastore.go]: Is there any sanitization for these attributes? I think we filter them in the controller which is good enough?

File: utils/inputs/pagination.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+		order      = map[string]bool{}
+	)
+	// allowed pagination ordering attributes
+	for _, o := range orderOptions {
+		order[o] = true
+	}
+	ctx = context.WithValue(ctx, appctx.PaginationOrderOptionsCTXKey, order)
+
+	if err := DecodeAndValidate(ctx, pagination, []byte(url)); err != nil {
+		fmt.Println("!!!!! err: ", err)
```
Comment (corymcdonald) [inline utils/inputs/pagination.go]: Should we log this error rather than printing?

File: docker-compose.yml [hunk 4]
```diff
@@ -87,6 +87,7 @@ services:
       - "BALANCE_SERVER=https://balance-staging.mercury.basicattentiontoken.org"
       - TOKEN_LIST
       - UPHOLD_ACCESS_TOKEN
+      - FEATURE_ORDERS=true
```
Comment (corymcdonald) [inline docker-compose.yml]: If this set to true will it honor the value in the environment?

File: utils/inputs/pagination.go (hunk 3 above)
Comment (husobee) [reply utils/inputs/pagination.go]: in next commit.

File: docker-compose.yml (hunk 4 above)
Comment (husobee) [reply docker-compose.yml]: reverted in next commit.

File: payment/datastore.go (hunk 2 above)
Comment (husobee) [reply payment/datastore.go]: line 79 is where the order parameters are sanitized, based on what is passed in to inputs.NewPaginiation variadic parameter.  Unfortunately with the order by statement, i don't think you can pass those in through the go libary?

Comment (corymcdonald) [reply payment/datastore.go:184]: No, don't think so either. I'm good with this 👍

**Pull request merged**